portfolio/src/main/java/com/google/sps/servlets/ReportServlet.java

```diff
@@ -0,0 +1,37 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that deletes comments data */
+@WebServlet("/report")
+public class ReportServlet extends HttpServlet {
+
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String id = request.getParameter("id");
+    try {
+      Key commentKey = KeyFactory.stringToKey(id);
+  		Entity commentEntity = datastore.get(commentKey);
+      long numReports = (long) commentEntity.getProperty("numReports");
+      numReports++;
+      commentEntity.setProperty("numReports", numReports);
+      datastore.put(commentEntity);
```

Comment:
Andrew in a different PR. Replace `like` with `report`
> What happens if two people have your website open and like the same comment at the exact same time? ;)

Reply:
I actually talked about this yesterday with Zoe. I feel like it's less important for my use case because I am counting report numbers and I arbitrarily chose 2 reports to be the number required for a comment to be taken down (it'd have to be more if the website had more traffic).
Nonetheless, I have amended my code according to this documentation: https://cloud.google.com/appengine/docs/standard/java/datastore/transactions#using_transactions
1) I accidentally amended onto the wrong commit (oops, forgot I did a merge, will make sure to check next time)
2) the documentation code uses an infinite loop and breaks from it, in class I've always learned that using infinite loops is bad practice, what is the truth? I feel like it might be less readable if I did numRetries in a for loop...

Reply:
In general, you want to have accurate data for future use cases. What if in the future you want to create a new page that displayed metrics that depend directly on the number of reports?

For your (2) comment, I agree. for loop is good. I would also put everything that inside the loop into a method called something like `incrementReportCount(...)`

Overall: Please don't spend much more time on this. It is mainly a good thing to think about for capstone.